### PR TITLE
fix mutate circular data structure and improve test case

### PR DESCRIPTION
**Pull request by wulianglongrd**
1. fix mutate: support circular data structure 
2. update the Demo's field name and cover struct private field test

### CODE REVIEW

File: pkg/fuzz/util.go [hunk 1]
```diff
@@ -137,16 +137,31 @@ func (h Helper) T() test.Failer {
...
+		// do not use t.Skip here, it will call runtime.Goexit()
+		t.Log(err.Error())
 	}
 }
 
-func mutateStruct(h Helper, e reflect.Value) error {
+func mutateStruct(ctx mutateCtx, e reflect.Value) error {
+	// this can happen when circular data structure
+	if ctx.curDepth >= ctx.maxDepth {
+		return fmt.Errorf("reach max depth and the mutation may not completed")
```
Comment (wulianglongrd) [inline pkg/fuzz/util.go:160]: If we return an error here, it will just be logged and ignored. Another option is to panic here

**Review by howardjohn — APPROVED** [review]

**Pull request merged**